quisp/modules/QRSA/HardwareMonitor/HardwareMonitor.cc

```diff
@@ -511,6 +577,144 @@ Matrix4cd HardwareMonitor::reconstruct_Density_Matrix(int qnic_id) {
   density_matrix_Y2.real() ).trace(); EV<<"Yerr = "<<Yerr_rate2<<"\n";*/
 }
 
+Matrix4cd HardwareMonitor::extended_reconstruct_Density_Matrix(int qnic_id, int partner) {
+  // II
+  double S00 = 1.0;
+  double S01 = (double)extended_tomography_data[qnic_id][partner]["XX"].plus_plus / (double)extended_tomography_data[qnic_id][partner]["XX"].total_count -
```

Comment:
make a variable and assign `extended_tomography_data[qnic_id][partner]` into it. it's too hard to read...
if you use a nested value more than twice, you should make a variable and assign it for readability

Reply:
Right. Let me fix that.